src/embed_tests/TestList.cs

```diff
@@ -0,0 +1,35 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using NUnit.Framework;
+using Python.Runtime;
+
+namespace Python.EmbeddingTest
+{
+    public class TestList
+    {
+        [OneTimeSetUp]
+        public void SetUp()
+        {
+            PythonEngine.Initialize();
+        }
+
+        [OneTimeTearDown]
+        public void Dispose()
+        {
+            PythonEngine.Shutdown();
+        }
+
+        [Test]
+        public void MissingGenericTypeTest()
+        {
+            Assert.Throws<PythonException>(() =>
+            PythonEngine.Exec($@"
+from System.Collections import IList
+IList[bool]
+"));
```

Comment:
There are a few problems with Python tests at the moment:
1. tests.pyproj is not part of pythonnet.sln, and has the old configurations still in it.
1. Python tests run under the current Python environment, which requires installing the development version of Python.NET into the current Python environment.  Anyone who forgets to do this will be testing the wrong version of Python.NET.
1. If Python.Runtime throws an exception (as in this case), the Visual Studio debugger will not be activated.
Using an embedded test avoids all of these problems.

Reply:
@tminka here's how I deal with this:
1. Open python tests code in VS Code in a separate window, or add tests.pyproj to .sln
2. `conda create -n pynet-dev-py39 python=3.9`
3. Open `Python.Runtime` project settings in VS, and create a new profile like this:
![image](https://user-images.githubusercontent.com/239520/102547454-1e11df00-406e-11eb-943f-8b82a7ae8e8f.png)

But true, these problems cause me to often prefer .NET tests over Python tests.

Maybe we should come up with a better way to run these tests from Visual Studio.

Reply:
1. Adding tests.pyproj to sln doesn't work right because of the old configurations in there.  It also changes the sln file, which will show up in the commit.
2. Creating a virtual environment doesn't really solve the problem.  You still need to update Python.NET within that environment.  Referring to that environment also changes tests.pyproj which changes the commit.
3. This runs but doesn't stop at any breakpoints or exceptions.  It won't stop even if you raise an error in Python.

Basically this is no replacement for the simplicity of cloning the repository, opening the sln in Visual Studio, building, and running tests there.

Reply:
I have found a solution that should make everyone happy.  The new test is a Python test, and there is a runner that can execute any Python test as an embedded test.

Reply:
Let's keep the runner in a separate PR. Meanwhile the test failures in CI have been fixed.